File: common/src/main/java/net/superricky/tpaplusplus/command/BackCommand.java
package net.superricky.tpaplusplus.command;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import net.minecraft.commands.CommandBuildContext;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;
import net.superricky.tpaplusplus.util.configuration.Config;
import net.superricky.tpaplusplus.util.manager.DeathManager;

import static net.minecraft.commands.Commands.literal;

public class BackCommand {
    public static void onRegisterCommandEvent(CommandDispatcher<CommandSourceStack> dispatcher, CommandBuildContext registry, Commands.CommandSelection selection) {
        dispatcher.register(literal(Config.BACK_COMMAND_NAME.get())
                .executes(context -> teleportToLastDeath(context.getSource())));
    }

    private static int teleportToLastDeath(CommandSourceStack source) throws CommandSyntaxException {
        DeathManager.teleportToLatestDeath(source.getPlayerOrException());
        return 1;
    }

    private BackCommand() {
    }
}

File: common/src/main/java/net/superricky/tpaplusplus/commands/back/BackCommand.kt
package net.superricky.tpaplusplus.commands.back

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.exceptions.CommandSyntaxException
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.superricky.tpaplusplus.commands.back.Back.teleportToLatestDeath
import net.superricky.tpaplusplus.config.Config

object BackCommand {
    fun onRegisterCommandEvent(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
            Commands.literal(Config.BACK_COMMAND_NAME.get())
                .executes { context: CommandContext<CommandSourceStack> ->
                    teleportToLastDeath(
                        context.getSource()
                    )
                }
        )
    }

    @Throws(CommandSyntaxException::class)
    private fun teleportToLastDeath(source: CommandSourceStack): Int {
        teleportToLatestDeath(source.playerOrException)
        return 1
    }
}

